src/lib/operators/print.cpp

```diff
@@ -0,0 +1,94 @@
+#include "print.hpp"
+
+#include <algorithm>
+#include <iomanip>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "operators/table_wrapper.hpp"
+#include "storage/base_segment.hpp"
+#include "storage/table.hpp"
+#include "type_cast.hpp"
+#include "utils/performance_warning.hpp"
+
+namespace opossum {
+
+Print::Print(const std::shared_ptr<const AbstractOperator> in, std::ostream& out) : AbstractOperator(in), _out(out) {}
+
+void Print::print(std::shared_ptr<const Table> table, std::ostream& out) {
+  auto table_wrapper = std::make_shared<TableWrapper>(table);
+  table_wrapper->execute();
+  Print(table_wrapper, out).execute();
+}
+
+std::shared_ptr<const Table> Print::_on_execute() {
+  PerformanceWarningDisabler pwd;
+
+  auto widths = column_string_widths(8, 20, _input_table_left());
+
+  // print column headers
+  _out << "=== Columns" << std::endl;
+  for (ColumnID column_id{0}; column_id < _input_table_left()->column_count(); ++column_id) {
+    _out << "|" << std::setw(widths[column_id]) << _input_table_left()->column_name(column_id) << std::setw(0);
+  }
+  _out << "|" << std::endl;
+  for (ColumnID column_id{0}; column_id < _input_table_left()->column_count(); ++column_id) {
+    _out << "|" << std::setw(widths[column_id]) << _input_table_left()->column_type(column_id) << std::setw(0);
+  }
+  _out << "|" << std::endl;
+
+  // print each chunk
+  for (ChunkID chunk_id{0}; chunk_id < _input_table_left()->chunk_count(); ++chunk_id) {
+    const auto& chunk = _input_table_left()->get_chunk(chunk_id);
+
+    _out << "=== Chunk " << chunk_id << " === " << std::endl;
+
+    if (chunk.size() == 0) {
+      _out << "Empty chunk." << std::endl;
+      continue;
+    }
+
+    // print the rows in the chunk
+    for (size_t row = 0; row < chunk.size(); ++row) {
+      _out << "|";
+      for (ColumnID column_id{0}; column_id < chunk.column_count(); ++column_id) {
+        // well yes, we use BaseSegment::operator[] here, but since Print is not an operation that should
```

Comment:
I am not sure if the instructors think that you should make this assumption, but I don't have anything against it.